lib/src/io.dart

```diff
@@ -553,10 +554,15 @@ Future<PubProcessResult> runProcess(String executable, List<String> args,
   ArgumentError.checkNotNull(executable, 'executable');
 
   return _descriptorPool.withResource(() async {
-    var result = await _doProcess(Process.run, executable, args,
-        workingDir: workingDir,
-        environment: environment,
-        runInShell: runInShell);
+    ProcessResult result;
+    try {
+      result = await _doProcess(Process.run, executable, args,
+          workingDir: workingDir,
+          environment: environment,
+          runInShell: runInShell);
+    } on IOException catch (e) {
+      throw ApplicationException('Failed to run subprocess `$executable`: $e');
```

Comment:
This is still an internal error...

I think we might have to tell people to file a bug, don't we?

Reply:
Even if it is internal the stack trace won't help much.

I think the exit code has to be checked for each invocation seperately. (And I believe we do that already).